### PR TITLE
Prioritization of SQL query work items

### PR DESCRIPTION
**Pull request by mistachkin**
These changes are designed to permit the prioritization of SQL work items based on a fixed number of criteria available within the context of the client request (e.g. originating user, associated task, originating machine, etc).

### CODE REVIEW

**Comment by nirbhayc** [comment]
Comdb2 logs for sysbench showed a couple of potential issues (not sure if those are related to this PR):
```
2019/07/16 18:56:11 [ERROR] bdb_cursor_open: no curtran provided!!!
2019/07/16 18:56:11 Comdb2's Linux Cheap Stack Trace :: pthread_self:139976862115584 @ /bb/comdb2a/roborivers/branches/master/bdb/cursor.c:581
...
2019/07/16 18:57:45 [ERROR] sqlite3_close_serial: sqlite3_close FAILED rc=5, msg=unable to close due to unfinalized statements or unfinished backups
2019/07/16 18:57:45 [ERROR] sqlite3_close_serial: sqlite3_close FAILED rc=5, msg=unable to close due to unfinalized statements or unfinished backups
```

File: db/comdb2_ruleset.c [hunk 1]
```diff
@@ -0,0 +1,1225 @@
...
+
+  rules->generation = ATOMIC_ADD64(gbl_ruleset_generation, 1);
+  *pRules = rules;
+
+  rc = 0;
+  goto done;
+
+failure:
+  logmsg(LOGMSG_ERROR, "%s", zError);
+  comdb2_free_ruleset(rules);
```
Comment (markhannum) [inline db/comdb2_ruleset.c]: HEY dont free entire ruleset just because we have a bug ..

**Comment by markhannum** [comment]
Looks great!  Okay to merge.

**Pull request merged**